packages/sanity/src/core/store/events/useEventsStore.ts

```diff
@@ -0,0 +1,199 @@
+import {type ObjectSchemaType} from '@sanity/types'
+import {useCallback, useEffect, useMemo} from 'react'
+import {useObservable} from 'react-rx'
+import {of} from 'rxjs'
+
+import {useClient, useSchema} from '../../hooks'
+import {useReleasesStore} from '../../releases/store/useReleasesStore'
+import {useWorkspace} from '../../studio/workspace'
+import {DEFAULT_STUDIO_CLIENT_OPTIONS} from '../../studioClient'
+import {getDocumentVariantType} from '../../util/getDocumentVariantType'
+import {fetchFeatureToggle} from '../_legacy/document/document-pair/utils/fetchFeatureToggle'
+import {createEventsStore} from './createEventsStore'
+import {getDocumentAtRevision} from './getDocumentAtRevision'
+import {
+  type DocumentGroupEvent,
+  type EventsStore,
+  isEditDocumentVersionEvent,
+  isPublishDocumentVersionEvent,
+} from './types'
+
+export interface EventsObservableValue {
+  events: DocumentGroupEvent[]
+  nextCursor: string
+  loading: boolean
+  error: null | Error
+}
+const INITIAL_VALUE: EventsObservableValue = {
+  events: [],
+  nextCursor: '',
+  loading: true,
+  error: null,
+}
+
+/**
+ * @internal
+ */
+export function useEventsStore({
```

Comment:
Yes, I think we need tests 100% agree on this. 
I suggested to add tests as a follow up given it will take me some time to get it right and I need to get into the release history. 
Will come back to this and add tests . I've written what I consider as a minimum to test in linear corel-280